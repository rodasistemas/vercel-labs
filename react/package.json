{
  "private": true,
  "dependencies": {
    "@material/react-material-icon": "^0.15.0",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.5.0",
    "material-icons": "^1.10.7",
    "node-sass": "^7.0.1",
    "react": "^17.0.2",
    "react-addons-test-utils": "^15.6.2",
    "react-app-rewired": "^2.2.1",
    "react-dom": "^17.0.2",
    "react-router-dom": "^6.2.2",
    "react-scripts": "^5.0.0",
    "typescript": "^4.6.2",
    "web-vitals": "^2.1.3"
  },
  "scripts": {
    "start": "react-app-rewired start",
    "build": "react-app-rewired build",
    "test": "react-app-rewired test",
    "eject": "react-app-rewired eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@types/react": "^17.0.39",
    "@types/react-dom": "^17.0.13",
    "msw": "^0.38.2",
    "react-app-rewire-alias": "^1.1.7"
  }
}
