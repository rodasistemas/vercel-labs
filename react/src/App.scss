body,
html {
    height: 100%;
    display: grid;
    position: fixed;
    margin: 0;
    padding: 0;
    width: 100%;
    overflow: hidden;
}

.App {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    margin: 0;

    padding: 0;
    position: relative;
    overflow: auto;
    &-header {
        background-color: #282c34;
        height: 70px;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        font-size: calc(10px + 2vmin);
        color: white;
    }
    &-container{
        margin:0;
        transition: margin 300ms ease-in-out 100ms;
        &.left{
            margin-left: 290px;
        }
    }
}
.App-sidebar {
    background-color: white;
    width: 260px;
    border-right: 1px solid white;
    box-shadow: 0 0 2px rgba(0, 0, 0, 0.5);
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    padding: 10px 5px;
    transition: all 300ms ease-in-out 100ms;
    position: fixed;
    margin: 0;
    &.left {
        left: 0;
    }
    &.right {
        right: 0;
    }
    &.sidebar {
        &-close {
            &.left {
                margin-left: -230px;
                + .App-container {
                    margin-left: 60px;
                }
                ul,li{
                    transition: all 300ms ease-in-out 100ms;
                    max-width: 10px;
                    opacity: 0;
                }
                + h3, .title {
                    opacity: 0;
                    text-align: left;
                }
            }
            &.right {
                margin-right: -230px;
            }
        }
    }
    ul {
        padding: 0;
        margin: 0;
        width: 100%;
        li {
            list-style: none;
            display: flex;
            border: 1px solid rgba(0, 0, 0, 0.1);
            padding: 8px;
            margin: 8px;
            border-radius: 5px;
            cursor: pointer;
            transition: all 100ms linear 10ms;
            &:hover{
                background-color:bisque ;
            }
        }
    }
    h3, .title {
        transition: all 300ms ease-in-out 100ms;
        display: flex;
        margin: 0;
        padding-top: 10px;
        padding-bottom: 24px;
        border-bottom: 1px solid rgba(0, 0, 0, 0.1);
        width: 100%;
        clear: both;
        align-items: center;
        justify-content: center;

    }

    & button {
        position: absolute;
        margin-top: auto;
        margin-bottom: auto;
        width: 30px;
        height: 30px;
        display: flex;
        border: 1px solid rgba(0, 0, 0, 0.1);
        padding: 0;

        align-items: center;
        justify-content: center;
        border-radius: 5px;
        cursor: pointer;
        color: gray;
        .material-icons {
            font-size: 18px;
        }
        :hover {
            color: royalblue;
            box-shadow: 0 1px 4px #282c34;
        }
        @at-root {
            .right {
                button {
                    right: auto;
                    left: 10px;
                }
            }
            .left {
                button {
                    left: auto;
                    right: 10px;
                }
            }
            .sidebar-close {
                button {
                    color: royalblue;
                    box-shadow: 0 1px 4px #282c34;
                }

                ul,
                h3 {
                    padding-left: 60px;
                }
            }
        }
    }
}
.App-logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .App-logo {
        animation: App-logo-spin infinite 20s linear;
    }
}

.App-link {
    color: #61dafb;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}
