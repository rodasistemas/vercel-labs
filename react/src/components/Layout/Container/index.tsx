const Container:React.FC<any> = (props) => {
    const { children} = props;
    return (
        <div className={`${props.className}`} role="container">
            {children}
        </div>
    )
}
export default Container;
