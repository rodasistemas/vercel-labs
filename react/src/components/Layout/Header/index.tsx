const Header: React.FC<any> = (props) => {
  const { children } = props
  return <header {...props}>{children}</header>
}
export default Header;

