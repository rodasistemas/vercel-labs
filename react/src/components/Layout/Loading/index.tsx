import { Spinner } from './spinner';

const Loading = () => {
    return (
        <div className="loading">
            <Spinner />
            </div>
    )
}
export default Loading;
