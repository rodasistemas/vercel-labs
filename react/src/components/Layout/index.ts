export {default as Header} from "./Header";

export { default as Sidebar } from "./Sidebar";

export { default as Container} from "./Container";

export {default as Loading} from "./Loading";
