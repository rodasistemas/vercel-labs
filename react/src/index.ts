import './index.css'
import App from './App'
import ReactDOM from 'react-dom'
import { createElement } from 'react'

ReactDOM.render(createElement(App), document.getElementById('root'))
