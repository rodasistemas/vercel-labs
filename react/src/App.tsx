import './App.scss';
import {Header, Sidebar} from "./components/Layout";

const App:React.FC<any> = () => {
  return (
    <div className="App">
      <Header className="App-header">

        React
      </Header>
      <Sidebar className="App-sidebar" title="Menu">
          <ul>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
              <li>Item</li>
          </ul>
      </Sidebar>
    </div>
  );
}

export default App;
