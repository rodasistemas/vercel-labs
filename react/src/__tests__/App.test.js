import React from 'react'
import {render, fireEvent, screen} from '@testing-library/react'
import '@testing-library/jest-dom'
import App from "../App"
import { Sidebar } from '../components/Layout'
/*
import {rest} from 'msw'
import {setupServer} from 'msw/node'
const server = setupServer(
    rest.get('/greeting', (req, res, ctx) => {
      return res(ctx.json({greeting: 'hello there'}))
    }),
  )

  beforeAll(() => server.listen())
  afterEach(() => server.resetHandlers())
  afterAll(() => server.close())
*/
  describe('[App]',()=>{
      it('render all page',()=>{
          render(<App/>)
          expect(screen.getByRole('banner')).toBeInTheDocument();
      })
      it('sidebar state',()=>{
          render(<Sidebar/>)
          expect(screen.getByRole('sidebar')).toHaveTextContent(/menu/i)
        })
        it('sidebar event',()=>{
          const {container} = render(<Sidebar className="test"/>);
          const buttonMock = screen.getByRole('button');
          const eventMock = fireEvent(buttonMock,new MouseEvent('click',{bubbles: true, cancelable:true}));
          expect(eventMock).toBeTruthy()
          expect(container).toMatchSnapshot()
      })
      it('snapshot',()=>{
          const {container} = render(<App/>);
          // eslint-disable-next-line testing-library/no-debugging-utils
          expect(container).toMatchSnapshot()
      })
  })
