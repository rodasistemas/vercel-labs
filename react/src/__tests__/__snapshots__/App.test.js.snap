// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[App] sidebar event 1`] = `
<div>
  <aside
    class="test sidebar-close"
    role="sidebar"
  >
    <h3
      class="sidebar-title"
    >
      <button
        class="sidebar-toggle"
      >
        <span
          class="material-icons"
        >
          menu
        </span>
      </button>
    </h3>
  </aside>
</div>
`;

exports[`[App] snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      AppLayout Header
    </header>
    <aside
      class="App-sidebar left sidebar-open"
      role="sidebar"
    >
      <h3
        class="sidebar-title"
      >
        AppLayout
        <button
          class="sidebar-toggle"
        >
          <span
            class="material-icons"
          >
            menu
          </span>
        </button>
      </h3>
      <ul>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
        <li>
          Item
        </li>
      </ul>
    </aside>
    <div
      class="App-container left"
      role="container"
    >
      AppLayout Container
    </div>
  </div>
</div>
`;
